llvm/lib/SYCL/InSPIRation/InSPIRation.cpp

```diff
@@ -404,8 +404,8 @@ struct InSPIRation : public ModulePass {
   /// InferAddressSpaces pass and teach it to deal with these SSDM calls as
   /// Intrinsics.
   void ssdmAddressSpaceFix(Function &F) {
-    for (inst_iterator I = inst_begin(F), E = inst_end(F); I != E; ++I)
-      if (auto *Call = dyn_cast<CallInst>(&*I))
+    for (auto &I : instructions(F))
```

Comment:
It is not possible to get directly the pointer with `auto *I`?

Reply:
It has some trouble deducing to auto *, and it's used with auto & in the other locations throughout LLVM that the instructions call is made. I've found that there's a couple of the Clang APIs that won't deduce auto *'s.